File: scripts/grant_temp_standard.js

import { createClient } from '@supabase/supabase-js';

const SUPABASE_URL = 'https://pbxrtqurwdfoplhbzyhd.supabase.co';
const SERVICE_ROLE_KEY = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InBieHJ0cXVyd2Rmb3BsaGJ6eWhkIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc2OTkwMTIwOSwiZXhwIjoyMDg1NDc3MjA5fQ.fqqIlI8KVpOKt6VFQXuj8lQPDSMf3kB6MBASThcDCAY';

const supabase = createClient(SUPABASE_URL, SERVICE_ROLE_KEY);

const TARGET_EMAILS = [
    'amitfrenkel20@gmail.com',
    'Tinofx17@gmail.com'
];

const DURATION_WEEKS = 52; // 1 Year

async function grantStandardAccess() {
    console.log(`🔍 GRANTING STANDARD ACCESS TO USERS`);
    console.log('='.repeat(60));

    try {
        // 1. Find Users by Email
        const { data: { users }, error: userError } = await supabase.auth.admin.listUsers();
        if (userError) throw userError;

        for (const email of TARGET_EMAILS) {
            const user = users.find(u => u.email.toLowerCase() === email.toLowerCase());

            if (!user) {
                console.error(`❌ User ${email} not found!`);
                continue;
            }

            console.log(`✅ Found user: ${user.email} (ID: ${user.id})`);

            // 2. Calculate Dates
            const startDate = new Date();
            const endDate = new Date();
            endDate.setDate(startDate.getDate() + (DURATION_WEEKS * 7));

            // 3. Update Profile
            const { error: updateError } = await supabase
                .from('profiles')
                .update({
                    subscription_status: 'active',
                    subscription_type: 'standard',
                    subscription_start_date: startDate.toISOString(),
                    subscription_end_date: endDate.toISOString(),
                    // Also set verification_status to verified so they don't get blocked by the new feature
                    verification_status: 'verified',
                    updated_at: new Date().toISOString()
                })
                .eq('id', user.id);

            if (updateError) {
                // If verification_status column doesn't exist yet, this might fail or just ignore it.
                // To be safe, let's try updating without verification_status first if it fails?
                // Actually, since I haven't added the column yet, it WILL fail if I include it.
                // GOOD CATCH. I should NOT include verification_status yet.
                // Re-running update without it.
                /*
                 Wait, Supabase ignores extra columns usually? Or throws error?
                 It usually throws error for columns that don't exist.
                 */
                console.warn("Attempting update without verification_status column...");
                const { error: retryError } = await supabase
                    .from('profiles')
                    .update({
                        subscription_status: 'active',
                        subscription_type: 'standard',
                        subscription_start_date: startDate.toISOString(),
                        subscription_end_date: endDate.toISOString(),
                        updated_at: new Date().toISOString()
                    })
                    .eq('id', user.id);

                if (retryError) throw retryError;
            } else {
                // If the first one succeeded (unlikely if col doesn't exist), great.
                // Actually the logic above is flawed (can't try/catch the result of the constant assignment).
            }

            // Let's just do the clean update since I haven't added the column yet.
            const { error: finalError } = await supabase
                .from('profiles')
                .update({
                    subscription_status: 'active',
                    subscription_type: 'standard',
                    subscription_start_date: startDate.toISOString(),
                    subscription_end_date: endDate.toISOString(),
                    updated_at: new Date().toISOString()
                })
                .eq('id', user.id);

            if (finalError) throw finalError;

            console.log(`✅ Successfully updated profile for ${email}`);
        }

    } catch (error) {
        console.error('❌ Error:', error.message);
    }
}

grantStandardAccess();

File: scripts/grant_temp_standard.py
import urllib.request
import json
import datetime
import ssl

# Bypass SSL verification if needed (like in fetch_dates.py)
ssl._create_default_https_context = ssl._create_unverified_context

SUPABASE_URL = 'https://pbxrtqurwdfoplhbzyhd.supabase.co'
# Using the Service Role Key from the original JS script
SERVICE_ROLE_KEY = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InBieHJ0cXVyd2Rmb3BsaGJ6eWhkIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc2OTkwMTIwOSwiZXhwIjoyMDg1NDc3MjA5fQ.fqqIlI8KVpOKt6VFQXuj8lQPDSMf3kB6MBASThcDCAY'

TARGET_EMAILS = [
    'amitfrenkel20@gmail.com',
    'Tinofx17@gmail.com'
]

HEADERS = {
    'apikey': SERVICE_ROLE_KEY,
    'Authorization': f'Bearer {SERVICE_ROLE_KEY}',
    'Content-Type': 'application/json'
}

def list_users():
    url = f"{SUPABASE_URL}/auth/v1/admin/users"
    req = urllib.request.Request(url, headers=HEADERS)
    try:
        with urllib.request.urlopen(req) as response:
            data = json.loads(response.read().decode('utf-8'))
            return data.get('users', [])
    except Exception as e:
        print(f"❌ Error listing users: {e}")
        return []

def update_profile(user_id, email):
    url = f"{SUPABASE_URL}/rest/v1/profiles?id=eq.{user_id}"
    
    start_date = datetime.datetime.now()
    end_date = start_date + datetime.timedelta(weeks=52) # 1 year
    
    payload = {
        "subscription_status": "active",
        "subscription_type": "standard",
        "subscription_start_date": start_date.isoformat(),
        "subscription_end_date": end_date.isoformat(),
        "updated_at": datetime.datetime.now().isoformat()
        # Not adding verification_status yet as column might not exist
    }
    
    data = json.dumps(payload).encode('utf-8')
    req = urllib.request.Request(url, data=data, headers=HEADERS, method='PATCH')
    
    try:
        with urllib.request.urlopen(req) as response:
            if response.status in [200, 204]:
                print(f"✅ Successfully updated profile for {email}")
            else:
                print(f"⚠️ Unexpected status {response.status} for {email}")
    except Exception as e:
        print(f"❌ Error updating profile for {email}: {e}")

def main():
    print("🔍 GRANTING STANDARD ACCESS (Python)")
    print("="*60)
    
    users = list_users()
    print(f"Found {len(users)} total users.")
    
    for email in TARGET_EMAILS:
        user = next((u for u in users if u.get('email', '').lower() == email.lower()), None)
        
        if user:
            print(f"✅ Found user: {email} (ID: {user['id']})")
            update_profile(user['id'], email)
        else:
            print(f"❌ User {email} not found!")

if __name__ == "__main__":
    main()
